src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -6,18 +6,21 @@
 
 import javafx.collections.ObservableList;
 import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.leave.UniqueLeaveList;
 import seedu.address.model.person.ComparablePerson;
 import seedu.address.model.person.Person;
 import seedu.address.model.person.UniquePersonList;
 import seedu.address.model.person.exceptions.PersonNotFoundException;
 
+
 /**
  * Wraps all data at the address-book level
  * Duplicates are not allowed (by .isSamePerson comparison)
  */
 public class AddressBook implements ReadOnlyAddressBook {
 
     private final UniquePersonList persons;
+    private final UniqueLeaveList leaves;
```

Comment:
In terms of design choices I believe we're separating the persons and leaves - AddressBook should only contain persons while LeavesBook should only contain leaves, and the ModelManager will contain an instance of AddressBook and LeavesBook